### PR TITLE
Activation metric for mobile new profiles

### PR DESCRIPTION
**Pull request by bani**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Comment by bani** [comment]
For this PR I was following the instructions from [Adding a new scheduled query](https://mozilla.github.io/bigquery-etl/cookbooks/common_workflows/#adding-a-new-scheduled-query), and a few things didn't work as expected.

Some of the bqetl commands seemed to complete successfully, but they always ended with `./bqetl: line 43: 92880 Segmentation fault: 11  venv/bin/bqetl "$@"`. I had this working previously, but it started happening when I had to update to newer Python version.

1. `./bqetl query create fenix.new_profile_activation_v1` ✅
2. Writing the query: I tested the query substituting `@submission_date` and it's working fine, but since this is my first query with incremental build it's worth double checking if it was done correctly
3. `./bqetl query validate fenix.new_profile_activation_v1` and `./bqetl query schema update fenix.new_profile_activation_v1` - for these commands I got `No files matching: fenix.new_profile_activation_v1`. In retrospect I should probably have used `fenix_derived` for the dataset, but that part wasn't clear in the documentation since for the first step I was using just fenix. With the schema being marked as optional, I skipped it.
4. Next creating the DAG, `./bqetl dag create bqetl_mobile_activation`. The result was informative, maybe the doc could just point out to run help to find out was needed:
```
Usage: bqetl dag create [OPTIONS] NAME
Try 'bqetl dag create --help' for help.

Error: Missing option '--schedule_interval' / '--schedule-interval'.
```

At this point I checked my git status and noticed two log files with Java errors. I can share if that's helpful. I tried to commit my changes and `yamllint` failed, the new section of the `dag.yml` file had missing indentation in a couple sections, which I manually fixed.

5. Then I tried the schedule, but again I was having issues with the wrong dataset name with `./bqetl query schedule fenix.new_profile_activation_v1 --dag bqetl_mobile_activation`. I ended up manually writing the config, so it needs extra checking.

Then I tried to repeat the steps to add the query for Firefox iOS. Some of that was manual because the folders didn't exist. Do I have the correct names of what they should be?

So I'm now at the step of having a PR with CI failing. I see that next I'd need to create the destination table, which requires the schema, which I previously skipped as it was marked optional. I'd like to get a review of how things are looking so far and double check if I should be using fenix_derived to attempt to auto-generate the schema.

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    app_build,
+    app_channel,
+    app_display_version,
+    device_manufacturer,
+    device_model,
+    isp,
+    locale,
+    city,
+    country,
+    MIN(first_seen_date) AS first_seen_date
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:15]: Is this needed? Are there sometimes multiple values for `first_seen_date` for a given `client_id`+`submission_date`?

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  USING
+    (client_id)
+  WHERE
+    submission_date
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 3 DAY)
+    AND @submission_date
+    AND normalized_app_name = 'Fenix'
+    AND DATE_DIFF(submission_date, first_seen_date, DAY)
+    BETWEEN 4
+    AND 7
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:51]: I think the linter made a weird choice here, wrapping the `BETWEEN` statements in parentheses ought to help with that

```suggestion
    (
      submission_date
      BETWEEN DATE_SUB(@submission_date, INTERVAL 3 DAY)
      AND @submission_date
    )
    AND normalized_app_name = 'Fenix'
    AND (
      DATE_DIFF(submission_date, first_seen_date, DAY)
      BETWEEN 4 AND 7
    )
```

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    5,
+    6,
+    7,
+    8,
+    9,
+    10,
+    11,
+    12
+),
+client_search AS (
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:34]: Just to make sure I understand this one, we're grabbing the number of searches by client_id that have occurred over the past 4 days, as long as the `first_seen_date` was on or before the start of that period?

Do we even need to do that second check (`DATE_DIFF(submission_date, first_seen_date, DAY) BETWEEN 4 and 7`)? You pulled from the `clients_first_seen` table of 6 days ago, so I think all values of `first_seen_date` will be at least 6 days ago?

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    client_id,
+    submission_date,
+    ARRAY_LENGTH(
+      mozfun.bits28.to_dates(mozfun.bits28.range(days_seen_bits, -5, 6), submission_date)
+    ) AS days_2_7,
+  FROM
+    `mozdata.fenix.baseline_clients_last_seen`
+  WHERE
+    submission_date = @submission_date
+    AND date_diff(submission_date, first_seen_date, DAY) = 6
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:66]: Here we're restricting to just the clients that were first seen 6 days ago, right?

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    ARRAY_AGG(metrics.string.first_session_adgroup)[SAFE_OFFSET(0)] AS adjust_adgroup,
+    ARRAY_AGG(metrics.string.first_session_campaign)[SAFE_OFFSET(0)] AS adjust_campaign,
+    ARRAY_AGG(metrics.string.first_session_creative)[SAFE_OFFSET(0)] AS adjust_creative,
+    MIN(DATE(submission_timestamp)) AS first_session_date
+  FROM
+    `mozdata.fenix.first_session`
+  WHERE
+    DATE(submission_timestamp)
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 6 DAY)
+    AND @submission_date
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:82]:
```suggestion
    (
      DATE(submission_timestamp)
      BETWEEN DATE_SUB(@submission_date, INTERVAL 6 DAY)
      AND @submission_date
    )
```

File: sql/moz-fx-data-shared-prod/firefox_ios_derived/new_profile_activation_v1/query.sql [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    (client_id)
+  WHERE
+    submission_date
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 3 DAY)
+    AND @submission_date
+    AND normalized_app_name = 'Fennec'
+    AND os = 'iOS'
+    AND DATE_DIFF(submission_date, first_seen_date, DAY)
+    BETWEEN 4
+    AND 7
```
Comment (curtismorales) [inline sql/moz-fx-data-shared-prod/firefox_ios_derived/new_profile_activation_v1/query.sql:52]:
```suggestion
    (
      submission_date
      BETWEEN DATE_SUB(@submission_date, INTERVAL 3 DAY)
      AND @submission_date
    )
    AND normalized_app_name = 'Fennec'
    AND os = 'iOS'
    AND (
      DATE_DIFF(submission_date, first_seen_date, DAY)
      BETWEEN 4 AND 7
    )
```

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql (hunk 1 above)
Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:15]: I'll remove. We have duplicates, but they are across different days, so now that I'm making this incremental I think I'll need to de-dup in the final query that will use this table.

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql (hunk 3 above)
Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:34]: Yes, I should be able to remove the second check

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql (hunk 4 above)
Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:66]: yes, this table is based on activity during their first 7 days so looking at the bit array at day 7 I can get the data.

File: sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql [hunk 7]
```diff
@@ -0,0 +1,123 @@
+WITH client_first_seen AS (
```
[the remark this answers is not in the document]
Comment (bani) [reply sql/moz-fx-data-shared-prod/fenix_derived/new_profile_activation_v1/query.sql:7]: added as a comment

**Review by curtismorales — APPROVED** [review]
Looks good! I'll run the deploy and make sure the build passes

**Pull request merged**